### PR TITLE
Superbuild: Pass additional ITK CMake configuration options

### PR DESCRIPTION
**Pull request by thewtex**
@fbudin69500 Could you please take this branch and run with it?

Todos:

- [ ] Test this patch, and make sure it works
- [x] Pass all `Module_*` options to the inner ExternalProject
<s> - [ ] Only enable `ITK_WRAP_DOC` if `DOXYGEN_EXECUTABLE` is available </s> pushed to Issue #116 
<s> - [ ] Do a better job at finding `DOXYGEN_EXECUTABLE` </s> pushed to Issue #116

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -153,6 +153,46 @@ if(ITKPythonPackage_SUPERBUILD)
...
+    ITK_WRAP_unsigned_short
+    ITK_WRAP_vector_double
+    ITK_WRAP_vector_float
+    )
+    if(DEFINED ${var})
+      list(APPEND ep_itk_cmake_cache_args "-D${var}=${${var}}")
+    endif()
+  endforeach()
+  # Todo, also pass all Module_* variables
+  message(STATUS "ITK CMake Cache Args -   ${ep_itk_cmake_cache_args}")
```
Comment (jcfr) [inline CMakeLists.txt]: You could simplify this by using https://cmake.org/cmake/help/v3.11/prop_dir/CACHE_VARIABLES.html

You could then automatically propagate all variables prefixed with either `ITK_WRAP_` or `Modules_` prefix

File: CMakeLists.txt [hunk 2]
```diff
@@ -153,6 +153,46 @@ if(ITKPythonPackage_SUPERBUILD)
   message(STATUS "SuperBuild - Searching for python[OK]")
   message(STATUS "SuperBuild -   DOXYGEN_EXECUTABLE: ${DOXYGEN_EXECUTABLE}")
 
+  # CMake configuration variables to pass to ITK's build
+  set(ep_itk_cmake_cache_args "")
+  foreach(var ITK_WRAP_DOC
```
Comment (jcfr) [inline CMakeLists.txt]: `ITK_WRAP_DOC` should be indented like the other variables

Sorting alphabetically also helps.

File: CMakeLists.txt (hunk 1 above)
Comment (fbudin69500) [reply CMakeLists.txt]: @jcfr The documentation says that this property is intended for debug purposes. Do you think it is a good idea to use it in this case?

Comment (jcfr) [reply CMakeLists.txt]: > this property is intended for debug purposes

This is a hint. It makes sense to use it here.

We should submit a MR to cmake to change this too.

With something like `this property is useful for debug purposes`

Other references:
* http://cmake.3232098.n2.nabble.com/Relaying-all-command-line-arguments-from-SuperBuild-to-ExternalProject-Add-tp7597038p7597045.html
* https://github.com/search?l=CMake&q=CACHE_VARIABLES&type=Code&utf8=%E2%9C%93

Comment (fbudin69500) [reply CMakeLists.txt]: Great. Thanks for your input.

**Comment by hjmjohnson** [comment]
@thewtex What's the status of this patch?

File: CMakeLists.txt (hunk 2 above)
Comment (thewtex) [reply CMakeLists.txt]: Done

**Comment by thewtex** [comment]
@hjmjohnson taking a look -- pushing an update that needs more testing.

**Comment by thewtex** [comment]
Updated and tested. Hats off :tophat: to @jcfr for the nice cache variable pattern matching suggestion.

**Pull request merged**